doc/howto/oidc_keycloak.md

```diff
@@ -0,0 +1,41 @@
+(oidc-keycloak)=
+# How to configure Keycloak as login method for the LXD UI
+
+Keycloak is a self-hosted open source tool for authentication. Keycloak supports OIDC and can be used to authenticate users for the LXD UI. This guide shows you how to set up Keycloak as the login method for the LXD UI.
+
+## Using Keycloak to access LXD
+
+1. Setup Keycloak, for this guide, it is assumed that Keycloak is available over HTTPS.
+   - Following their guide on [configuring Keycloak for production](https://www.keycloak.org/server/configuration-production).
+   - Alternatively run the development version: Download [Keycloak-25.0.4](https://github.com/keycloak/keycloak/releases/download/25.0.4/keycloak-25.0.4.zip), extract the file and run `bin/kc.sh start-dev`. Open `http://localhost:8080/` and create an admin user with password.
+
+1. Sign in to Keycloak with an admin account. Select the {guilabel}`Keycloak` dropdown in the top left corner of the admin console. Click {guilabel}`Create realm`. Enter a {guilabel}`Realm name` such as `lxd-ui-realm` and click {guilabel}`Create`.
+
+1. Go to {guilabel}`Realm Settings` on the newly created realm. Enter the IP or domain for Keycloak as {guilabel}`Frontend URL`, such as `http://192.0.2.1:8080`. Click {guilabel}`Save`.
```

Comment:
What is this for?

Reply:
Without this, the login dance fails on redirect from lxd to keylcoak. I suspect this is a security setting within keycloak.

Reply:
Interesting, I've never experience that. Did you deploy it behind a load-balancer or something? I've only ever run it with podman

Reply:
Without this, during the UI login process I get `{"type":"error","status":"","status_code":0,"operation":"","error_code":500,"error":"Login failed: Failed to get OIDC relying party: OpenID Provider Configuration Discovery has failed\nGet \"http://localhost:8080/realms/lxd-ui-realm/.well-known/openid-configuration\": Unable to connect to: localhost:8080 ([dial tcp 127.0.0.1:8080: connect: connection refused])","metadata":null}
`

Reply:
You are right, this is not needed.

Reply:
Could still be useful to add as a note though if it depends on the deployment configuration :)